### ISSUE TITLE
Incorrect XSS test in #2597

### ISSUE THREAD

**Issue by Conduitry** [sveltejs/kit#2648]
### Describe the bug

The test added along with the XSS fix in #2597 is not correct. Besides the `pwned`/`pnwed` spelling discrepancy noted in a comment on there, there appear to be other issues as well, as I'm not getting the tests to fail properly when I revert the rest of the XSS fix.

### Reproduction

Revert the fix part of #2597 and run the tests.

### System Info

```shell
n/a
```

### Severity

serious, but I can work around it

**Comment by baseballyama** [comment]
> (note: should it be a double await because page.evaluate itself returns a promise, so are we dealing with a promise of a promise here?)

I executed a promise of a promise on the playground.
At least below repl code, It seems like that we don't need double await.
(Please point it out if something is wrong.)

https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABBAFjANgEwPoAcBOcAtjAM4CmAFAJQBciACoSRQDwBucMmAfIgN4BYAFCIxifOSgh8SMOQDujZmSqVJpOOnbkANBPIArctGqIAvHyGjxtiAk3pyAOnRwA5pQBEpKAEN8KHpUDBwCYlUvagBuEVtbDS0dGlibeOQHLRc3Ty9yMExgtCw8FQoo1NsAXxiRKpERUEhYBERcAPyoUoiKShCS8JZyehoLPiYe8g4uXjplSdYJoenuHis48UlpWUR5JSXVSnVyRx19SWNTMYEN9PswR2yPb18AoLaOsC7ByNq0+MS2io-TCZSo1D+6TE90ermeeQK9Haki+3SGFVuiBqqXqwhEflIAE9IIgmtB4EgiH4YGBRtY7JknHDcq9AlFMX4FNSoB8Ud8wX1iqDJpDxDCsszvPlMOzhLiGsIqTSUkA

But `ambient.d.ts` defined `window.started` as boolean but actual type is `Promise<void>`.
We might want to fix that.

**Comment by benmccann** [comment]
The test being referred to is `no xss via query param` (https://github.com/sveltejs/kit/pull/2597/files#r732116874). I just inserted a couple extra characters at the beginning of the `key` query parameter and the test still passed. So it still seems this test isn't actually testing much of anything

> But i did verify the timeout only happens when xss is injected without the fix in place, so even if it is not the correct message the test does identify xss happening.

@dominikg I'm not so sure about this. If it's true I'd be happy to close this, but I'd expect the test to fail if I add extra characters to `key`?

>pwned/pnwed spelling discrepancy

This has been fixed in `master`

>By the way, ambient.d.ts defined window.started as boolean but actual type is Promise<void>.
We might want to fix that.

This has been fixed in `master`
